src/Http/Controllers/AuthorizationController.php

```diff
@@ -31,17 +34,28 @@ class AuthorizationController
      */
     protected $response;
 
+    /**
+     * The guard implementation.
+     *
+     * @var \Illuminate\Contracts\Auth\StatefulGuard
+     */
+    protected $guard;
+
     /**
      * Create a new controller instance.
      *
      * @param  \League\OAuth2\Server\AuthorizationServer  $server
      * @param  \Illuminate\Contracts\Routing\ResponseFactory  $response
+     * @param  \Illuminate\Contracts\Auth\StatefulGuard  $guard
      * @return void
      */
-    public function __construct(AuthorizationServer $server, ResponseFactory $response)
+    public function __construct(AuthorizationServer $server,
+                                ResponseFactory $response,
+                                StatefulGuard $guard)
```

Comment:
Do we resolve StatefulGuard like this anywhere else in Passport? How do we know this is the right guard or is even valid?

Reply:
No we don't, what if I add the following to `register` method of the `PassportServiceProvider`?
```php
$this->app->bind(StatefulGuard::class, function () {
    return Auth::guard();
});
```

Reply:
@driesvints Can you assist on this, please?